include/spi/core/messages.hpp

```diff
@@ -0,0 +1,106 @@
+#pragma once
+
+#include "common/core/message_queue.hpp"
+#include "spi/core/utils.hpp"
+
+namespace spi {
+
+namespace messages {
+
+/**
+ *
+ * @brief Transaction identifier struct
+ *
+ * @param[token] id of the message (generally the register reading/writing to)
+ * @param[command_type] whether it's a read or write command
+ * @param[requires_response] whether the message requires a response
+ */
+struct TransactionIdentifier {
+    uint8_t token;
```

Comment:
probably a good idea to make this a little bigger; it can be used as a bitmap by calling tasks to differentiate different kinds of calls at a higher level, and it's nice to have the space. uint32 maybe?